tests/rptest/tests/compatibility/flink_test.py

```diff
@@ -0,0 +1,122 @@
+from rptest.services.compatibility.flink_cluster import FlinkCluster
+import rptest.services.compatibility.flink_examples as FlinkExamples
+from rptest.services.kaf_producer import KafProducer
+from rptest.services.rpk_consumer import RpkConsumer
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.types import TopicSpec
+import math
+
+
+class FlinkTest(RedpandaTest):
+    """
+    Base class for Flink tests that contains all
+    shared objects between tests and starts the
+    Flink cluster
+    """
+
+    # The JOB is the java program that uses Flink with kafka connectors.
+    # The java program is represented by a wrapper in FlinkExamples.
+    JOB = None
+
+    # The producer should be an extension to KafProducer
+    PRODUCER = None
+
+    def __init__(self, test_context):
+        super(FlinkTest, self).__init__(test_context=test_context)
+
+        self._ctx = test_context
+
+        self._max_records = 10
+        self._timeout = 120
+
+        self._flink_cluster = FlinkCluster(test_context)
+
+    def is_valid_msg(self, msg):
+        raise NotImplementedError("is_valid_msg() is undefined")
+
+    def setUp(self):
+        self._flink_cluster.start()
+
+        super().setUp()
+
+    @cluster(num_nodes=6)
+    def test_flink(self):
+        # This will raise TypeError if JOB is undefined
+        flink_job = self.JOB(self.redpanda)
+
+        # This will raise TypeError if PRODUCER is undefined
+        producer = self.PRODUCER(self._ctx, self.redpanda, self.topics[0].name,
+                                 self._max_records)
+        consumer = RpkConsumer(self._ctx, self.redpanda, self.topics[1].name)
+
+        # Start the flink job on the
+        # cluster node
+        self._flink_cluster.run_flink_job(flink_job, self._timeout)
+
+        # Produce some messages
+        producer.start()
+        producer.wait()
+
+        # Consume the data
+        consumer.start()
+
+        def try_cons():
+            i = 0
+            msgs = consumer.messages
+            while i < len(msgs) and not self.is_valid_msg(msgs[i]):
+                i += 1
+
+            return i < len(msgs)
+
+        wait_until(try_cons,
+                   timeout_sec=self._timeout,
+                   backoff_sec=5,
+                   err_msg=f"{self._ctx.cls_name} consumer failed")
+
+        consumer.stop()
+        producer.stop()
+
+
+class WordProducer(KafProducer):
+    def __init__(self, context, redpanda, topic, records):
+        super(WordProducer, self).__init__(context,
+                                           redpanda,
+                                           topic,
+                                           num_records=records)
+
+    def value_gen(self):
+        return "\"redpanda is fast\""
+
+
+class FlinkWordCount(FlinkTest):
+    """
+    Test Flink with Kafka using a simple streaming job that consumes
+    from an input topic
+    """
+    topics = (
+        TopicSpec(name="input-topic"),
```

Comment:
I'm guessing these topic names are set to match hardcoded ones from the java example code?  If so would be good to note that in a comment so that nobody tries to change them later.